Updates to documentation (#117)
sodascience/metasynth: docs/source/index.rst

   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

.. image:: /images/logos/blue.svg
   :alt: MetaSynth_Logo

MetaSynth Documentation
=======================

> Welcome to the MetaSynth documentation. MetaSynth is a Python package that facilitates the generation of synthetic data and metadata, enabling users to create surrogate datasets that mimic the structure and characteristics of their original data while preserving privacy.

Reviewer: It would be nice to have a link to the GitHub page somewhere on this page.

Author: Latest version embeds a link in the first 'MetaSynth', I will also add an explicit link in the 'about' page.

.. warning:: 
   MetaSynth, and this documentation, are under active development. As a result, this documentation is still heavily work in progress. If you see any errors, missing content or have a suggestion for new content, feel free to :doc:`let us know </about/contact>`! 

Documentation Outline
---------------------

This documentation is designed to help you easily navigate and find the information you need. It is organized into the following four sections:

About
^^^^^
This section provides a brief overview of MetaSynth's purpose and functionality, contact information, frequently asked questions (FAQs), and licensing details.

> .. toctree::

Reviewer: The tree is already on the left panel. I think this section would be much more legible without the toctrees.

Author: I am not sure about that, let's discuss in person. I do agree that it clutters it visually, but I do think it aids a lot with overall ease of navigation and ease of understanding the documentation's structure. 

Main benefit is, that when people are looking up the documentation on a narrow screen (phones for example), the sidebar doesn't appear by default. In this scenario, having the TOC's in the text increases the ease of navigation (especially for first time browsers) of the docs. On top of that, the sidebar is not folded out by default, so the toctrees on the page still add information to the reader.

   :maxdepth: 2

   about/about

> User's Guide

Reviewer: Maybe "User guide"?

Author: I took inspiration from other documentation pages that used this terminology, that said, I am open to both.

> User's Guide

Reviewer: It would be nice if these were clickable.

Author: Agreed!

^^^^^^^^^^^^

> This section is a practical manual on how to use MetaSynth. It contains detailed, step-by-step instructions, as well as best practices for using the package. If you're new to MetaSynth, we recommend you start here!

Reviewer: ```suggestion
The user guide contains detailed, step-by-step instructions, as well as best practices for using MetaSynth. If you're new to MetaSynth, we recommend you start here!
```

.. toctree::
   :maxdepth: 2

   usage/usage

> API Reference

Reviewer: This should come second last before the About section.

Author: I don't think I understand this comment, could you elaborate?

^^^^^^^^^^^^^

> This section delves into the technical specifics of MetaSynth. Here, each function, class, and module is outlined in detail, giving you a comprehensive understanding of how the package works and how to use its various functionalities.

Reviewer: ```suggestion
This section is a technical reference for MetaSynth. Here, each function, class, and module is outlined in detail, giving you a comprehensive understanding of how the package works and how to use its various functionalities. If you want for example to understand which parameters can be used for which function, then you can find it here.
```

Author: ✅

This approach ensures the synthetic data remains separate and independent from any sensitive source data. Researchers and data owners can use MetaSynth to generate and share synthetic versions of their sensitive datasets, mitigating privacy concerns. Furthermore, the separation of metadata from original data promotes reproducibility, as the shareable metadata can be used to generate consistent synthetic data.

.. note:: 
   For more information on MetaSynth and its features, check out :doc:`/about/what_is`.

.. warning:: 
   MetaSynth, and this documentation, are under active development. As a result, this documentation is still heavily work in progress. If you see any errors, missing content or have a suggestion for new content, feel free to :doc:`let us know </about/contact>`! 

Documentation Outline
---------------------

.. toctree::
   :hidden:
   :maxdepth: 2

   about/about
   usage/usage
   developer/developer
   api/metasynth
   faq

This documentation is designed to help you easily navigate and find the information you need. It is organized into the following four sections:

:doc:`/about/about`
^^^^^^^^^^^^^^^^^^^

> The :doc:`About Section </about/about>` provides a brief overview of MetaSynth's purpose and functionality, contact information, frequently asked questions (FAQs), and licensing details.

Reviewer: The FAQ has moved, right, so maybe this needs to be updated?
